### PR TITLE
Apostrophe and quote consistency

### PR DESCRIPTION
**Pull request by cvalarida**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/4321

This theoretically changes all apostrophes and quotation marks to use the "curly" versions. There's a bunch here.

Here's how I did it:
Step to replace the `'`s (duplicated in `content/` and `src/js/`):
1) Use the regex `(\w+)\\?'(s|ve|re|d|ll|t|m)` to find contractions and replace the with `$1’$2`
2) Use the regexes `("[^"]*s\\?)'([^"]*")` and `('[^']*s)\\'([^']*')` to find dangling apostrophes (plural possessive `Veterans'`) and replace with `$1’$2`
2) Run `npm run test:unit` to make sure nothing broke.
3) Fix the lint errors (strings need single quotes)

Steps to replace the `"`s (duplicated in `content/` and `src/js/`):
1) Replace the `"`s using the following regexes with `!!quote!!`
    - `(={?[\s\n]*)"([^"]*)"` --> `$1!!quote!!$2!!quote!!`
      - This looks for string assignments--like `class="something"`
      - Also for JSX property assignments--like `serviceRequired={"messaging"}`
    - `((?:{%|{{)[^"]*)"([^"]*)"([^"]*(?:%}|}}))` --> `$1!!quote!!$2!!quote!!$3`
      - This looks for the liquid templating engine magic strings `{{ ... }}` and `{% ... %}`
      - Predictably, this has no hits in `src/js/`
    - `"([^"]*)"(\s?:)` --> `!!quote!!$1!!quote!!$2`
      - This looks for variable assignments inside objects (property name): `{ "propertyName": true }`
    - `(:\s*)"([^"]*)"` --> `$1!!quote!!$2!!quote!!`
      - This looks for variable assignments inside objects (value): `{ propertyName: "foo" }`
2) Find all `"([^"]*)"`s and change to `“$1”`
    - In `src/js/`, I actually went through the 180ish results manually because this also matched a bunch of comments and some regex validations that would have been bad to mess up
      - ...and other things we need the good ole `"` for
    - I only changed 3 instances in `src/js/`
3) Find all the `"`s again to see if we missed any inception quotes
4) Replace all `!!quote!!`s with `"`
    - 54570 replacements in 287 files (`content/`)
5) Check out any `.pdf`s that were unintentionally modified
6) Run unit tests to ensure nothing is too broken

I feel like my regex skill leveled up today, but there's probably a better way to do it then temporarily replacing all the `"`s I didn't want to mangle. Anyhow, there's a **lot** of changes here, so we may want to get more eyes reviewing.

### CODE REVIEW

**Review by cehsu — APPROVED** [review]
This goods good to me. It looks like `m` should be added to the contractions regex logic, as there are a few "I'm"'s such as in `content/pages/health-care/about-va-health-care/va-health-care-and-other-insurance.md`.

**Comment by cvalarida** [comment]
Good catch, @cehsu! Changes made, original comment edited. Thanks!

**Review by peggygannon — CHANGES_REQUESTED** [review]
@cvalarida Hello! On the following pages, it looks like double quotes were changed to single quotes or single quotes were changed to double. Can you take a look at these? 
src/js/login/containers/Main.jsx

src/js/messaging/components/ErrorView.jsx

src/js/rx/components/ErrorView.jsx

test/common/schemaform/SaveInProgressErrorPage.unit.spec.jsx

test/disability-benefits/e2e/03.claim-details.e2e.spec.js

**Comment by cvalarida** [comment]
Hey @peggygannon. Thanks for the detailed review!

Those `"`s turning into `'`s is fine. Actually, it's preferred. So previously, there were `'`s inside `"`s, but now that those `'`s are `’`, we can turn those `"`s into `'`s for consistency in our codebase.

**Comment by peggygannon** [comment]
@cvalarida Hello. Here is the list of pages that I have a question  if the quote change from straight to curly will break anything, as some of these pages do not look to be user-facing. 

src/js/auth/auth-entry.jsx

src/js/burials/config/form.js

src/js/burials/helpers.jsx

 src/js/common/components/LoginModal.jsx

 src/js/common/components/SignInLink.jsx

src/js/common/components/SortableTable.jsx

src/js/common/components/form-elements/ErrorableCurrentOrPastDate.jsx

src/js/common/components/form-elements/ErrorableNumberInput.jsx

src/js/common/components/form-elements/ErrorableTextarea.jsx

src/js/common/components/form-elements/GrowableTable.jsx

src/js/common/schemaform/ArrayField.jsx

src/js/common/schemaform/FormApp.jsx

src/js/common/schemaform/FormNav.jsx

src/js/common/schemaform/FormPage.jsx

src/js/common/schemaform/FormSaved.jsx

src/js/common/schemaform/ObjectField.jsx

src/js/common/schemaform/SchemaForm.jsx

src/js/common/schemaform/actions.js

src/js/common/schemaform/definitions/address.js

src/js/common/schemaform/definitions/file.js

src/js/common/schemaform/formState.js

src/js/common/schemaform/helpers.js

src/js/common/schemaform/reducers/index.js

src/js/common/schemaform/review/ArrayField.jsx

src/js/common/schemaform/review/ReviewCollapsibleChapter.jsx

src/js/common/schemaform/review/ReviewPage.jsx

src/js/common/schemaform/save-load-actions.js

src/js/common/schemaform/widgets/ArrayCountWidget.jsx

src/js/common/utils/helpers.js

src/js/edu-benefits/1990-rjsf/config/form.js

src/js/edu-benefits/1995/helpers.jsx

 src/js/edu-benefits/5490/helpers.jsx

src/js/edu-benefits/definitions/toursOfDuty.js

src/js/edu-benefits/pages/applicantService.js

src/js/facility-locator/components/ResultsList.jsx

src/js/facility-locator/containers/VAMap.jsx

src/js/gi/containers/SearchPage.jsx

src/js/hca/definitions/child.js

src/js/hca/helpers.jsx

src/js/letters/actions/letters.js

src/js/messaging/actions/messages.js

src/js/messaging/components/ThreadHeader.jsx

 src/js/messaging/containers/Folder.jsx

src/js/messaging/containers/Thread.jsx

src/js/messaging/reducers/messages.js

src/js/pensions/config/form.js

c/js/pensions/helpers.jsx

 src/js/rx/containers/Active.jsx

**Comment by cvalarida** [comment]
@peggygannon Thanks again for such a detailed review! I looked through all those files and they're good to go. There were a couple that were user-facing, but the rest are in comments. Ideally, I wouldn't change those at all, but it's more of a pain to skip them or revert them, so I didn't bother.

Also, since you requested a change, I can't merge this until you give it your approval.

**Review by peggygannon — APPROVED** [review]
@cvalarida Thanks for double-checking those pages. This PR is approved.

File: content/pages/health-care/about-va-health-care/va-health-care-and-other-insurance.md
Comment (cvalarida) [inline content/pages/health-care/about-va-health-care/va-health-care-and-other-insurance.md]: I just used what was in `master`, hoping this was right. Not sure who to ping on this one.

File: src/js/letters/containers/Main.jsx
Comment (cvalarida) [inline src/js/letters/containers/Main.jsx]: Again, I used what was in `master` and hoped for the best.

cc @annekainicUSDS to verify this is correct.

File: src/js/post-911-gib-status/utils/helpers.jsx
Comment (cvalarida) [inline src/js/post-911-gib-status/utils/helpers.jsx]: `master` didn't have these. I don't know if that's right or not, but I stuck with removing them.

Again, not sure who to ping on this. @bshyong or @U-DON?

**Review by cvalarida — COMMENTED** [review]
There were some funny merge conflicts that I _think_ I resolved correctly (by using what was in `master`), but I'd like verification.

File: src/js/post-911-gib-status/utils/helpers.jsx
Comment (annekainicUSDS) [inline src/js/post-911-gib-status/utils/helpers.jsx]: Yep, that's correct @cvalarida! I moved them into a common helper file.

File: src/js/letters/containers/Main.jsx
Comment (annekainicUSDS) [inline src/js/letters/containers/Main.jsx]: Yes this looks good!

**Pull request merged**
